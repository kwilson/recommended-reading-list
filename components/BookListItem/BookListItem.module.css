.container {
  align-items: flex-start;
  display: flex;
  flex-direction: column;
  margin-bottom: 0.5rem;
  border-bottom: dashed 1px var(--border-color);
  margin-bottom: 1rem;
  padding-bottom: 1rem;
  flex-grow: 1;
}

@media screen and (min-width: 460px) {
  .container {
    align-items: center;
    flex-direction: row;
  }
}

.cover {
  margin-right: 1rem;
  order: 0;
}

.details {
  order: 1;
}

.coverImage {
  border: solid 1px var(--border-color);
  max-width: 120px;
  height: auto;
}

.title {
  margin: 0;
}

.link {
  margin-top: 0.2rem;
}
