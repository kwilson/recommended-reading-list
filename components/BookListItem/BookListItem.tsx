import { FunctionComponent } from 'react';
import Link from 'next/link';

interface IProps {
  author: string;
  cover: string;
  link: string;
  title: string;
}

import styles from './BookListItem.module.css';

export const BookListItem: FunctionComponent<IProps> = ({
  author,
  cover,
  title,
  link,
}) => (
  <div className={styles.container}>
    <div className={styles.details}>
      <h2 className={styles.title} data-testid="book-title">
        {title}
      </h2>
      <div className={styles.author}>
        by <span data-testid="book-author">{author}</span>
      </div>

      <div className={styles.link} data-testid="book-link">
        <Link href={link}>More Info</Link>
      </div>
    </div>

    <div className={styles.cover}>
      <img
        data-testid="book-image"
        src={cover}
        alt={`${title} book cover`}
        className={styles.coverImage}
      />
    </div>
  </div>
);
