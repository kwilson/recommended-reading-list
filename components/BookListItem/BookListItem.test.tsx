import { render, screen } from '@testing-library/react';

import { BookListItem } from './BookListItem';

describe(`<${BookListItem.name} />`, () => {
  it('renders as expected', async () => {
    const props = {
      author: 'author value',
      cover: 'cover value',
      title: 'title value',
      link: 'link value',
    };

    const { asFragment } = render(<BookListItem {...props} />);

    expect(screen.queryByTestId('empty-book-list')).toBe(null);

    expect(screen.getByTestId('book-title').textContent).toEqual(props.title);
    expect(screen.getByTestId('book-author').textContent).toEqual(props.author);

    expect(screen.getByTestId('book-image').getAttribute('src')).toEqual(
      props.cover
    );
    expect(screen.getByTestId('book-image').getAttribute('alt')).toEqual(
      expect.stringContaining(props.title)
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
