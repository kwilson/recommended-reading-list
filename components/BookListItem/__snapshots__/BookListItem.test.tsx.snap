// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookListItem /> renders as expected 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="details"
    >
      <h2
        class="title"
        data-testid="book-title"
      >
        title value
      </h2>
      <div
        class="author"
      >
        by 
        <span
          data-testid="book-author"
        >
          author value
        </span>
      </div>
      <div
        class="link"
        data-testid="book-link"
      >
        <a
          href="/link%20value"
        >
          More Info
        </a>
      </div>
    </div>
    <div
      class="cover"
    >
      <img
        alt="title value book cover"
        class="coverImage"
        data-testid="book-image"
        src="cover value"
      />
    </div>
  </div>
</DocumentFragment>
`;
